src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs

```diff
@@ -190,6 +190,15 @@ internal sealed partial class ConsoleHost
 
                 s_cpp.Parse(tempArgs);
 
+                if (s_cpp.ShowVersion)
+                {
+                    // Alternatively, we could call s_theConsoleHost.UI.WriteLine(s_theConsoleHost.Version.ToString());
+                    // or start up the engine and retrieve the information via $psversiontable.GitCommitId
+                    // but this returns the semantic version and avoids executing a script
+                    s_theConsoleHost.UI.WriteLine("powershell " + PSVersionInfo.GetCommitInfo());
```

Comment:
I recommend to use `PSVersionInfo.GetPSVersionTable()["GitCommitId"]`, or even better, add an internal static property `GitCommitId` right after the `PSVersion` property at [here](https://github.com/PowerShell/PowerShell/blob/master/src/System.Management.Automation/engine/PSVersionInfo.cs#L148), and then use `PSVersionInfo.GitCommitId` here.

Reason: PSVersionInfo has a type initializer, which calls `GetCommitInfo()` and store the value in a hashtable. the current implementation of `GetCommitInfo()` reads git commit information from a file, which is slow and this implementation is subject to change (this method may be gone in near future -- we want to generate code at build time to embed the git commit id info, so as to avoid reading a file at startup time). Since the type initializer must run the first time you access a type, it's a waste to call `GetCommitInfo()` here to read the file again.